doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+ip link set dev vrrp6-2-1 up
+```
+The created interfaces will look like this:
+```
+$ ip addr show vrrp4-2-1
+5: vrrp4-2-1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
+    link/ether 00:00:5e:00:01:05 brd ff:ff:ff:ff:ff:ff
+    inet 10.0.2.16/24 scope global vrrp4-2-1
+       valid_lft forever preferred_lft forever
+    inet6 fe80::dc56:d11a:e69d:ea72/64 scope link stable-privacy
+       valid_lft forever preferred_lft forever
+
+$ ip addr show vrrp6-2-1
+8: vrrp6-2-1@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
+ link/ether 00:00:5e:00:02:05 brd ff:ff:ff:ff:ff:ff
+ inet6 2001:db8::370:7334/64 scope global
+    valid_lft forever preferred_lft forever
+ inet6 fe80::f8b7:c9dd:a1e8:9844/64 scope link stable-privacy
+    valid_lft forever preferred_lft forever
+```
+a few things to note about this interface:
+- It is slaved to eth0; any packets transmitted on this interface will egress via eth0
+- Its MAC address is set to the VRRP virtual MAC specified by the RFC for VRID 5
+- The VIP address 10.0.2.16 must not be present on the parent interface eth0
+- The link local address on the interface is not derived from the interface MAC
+
+#### Functional Description
+
+##### Protocol version selection
+
+Optionally specify the protocol version(VRRPv2 or VRRPv3). If the protocol version is not specified, the default is VRRPv3. Note that VRRPv2 does not support IPv6, so any IPv6 configuration will be rejected by FRR when using VRRPv2.
+
+##### Pre-emption
+
+Toggle preempt mode. When enabled, preemption allows Backup routers with higher priority to take over Master status from the existing Master. Enabled by default.
+
+##### Virtual Router Owner
+
+VRRP instance whose virtual IP address (VIP) is same as real interface address is called the owner of virtual router and has the highest priority (255). VRRP owner is supported in SONIC's VRRP implementation. VRRP owner node will pre-empt the active master even when pre-emption is disabled for that particular VRRP instance.
+
+##### Virtual MAC Address
+
+Following virtual MAC address is used by the protocol (as per RFC 5798)
+
+IPv4 case: 00-00-5e-00-01-{vrid}
+
+IPv6 case: 00-00-5e-00-02-{vrid}
+
+where, vrid is user configured 1-byte virtual router identifier. VRID has interface scope; that is, VRID has to be unique among the VRRP instances on an interface.
+
+##### VRRP Advertisement Frame
+
+VRRP control packets have IP protocol type as 112 (reserved for VRRP), for IPv4 network, are sent to VRRP multicast address 224.0.0.18, source MAC in VRRP control packets is virtual MAC address and source IP is interface IP (not the virtual IPv4 address). For IPv6 networks, the source address in the VRRPv6 header is the IPv6 link local address of the sending interface, and the destination address is the IPv6 multicast address FF02:: 02/64 assigned by IANA.
+
+##### Uplink Interface Tracking
```

Comment:
I believe, this uplink interface tracking can be a standalone feature instead of combining as part of VRRP.

Reply:
yes, you are right, will consider to exact it out for a new feature "monitor link" in phase 2.